sql/catalyst/src/main/java/org/apache/spark/sql/execution/RecordBinaryComparator.java

```diff
@@ -42,16 +42,16 @@ public int compare(
       while ((leftOff + i) % 8 != 0 && i < leftLen) {
         res = (Platform.getByte(leftObj, leftOff + i) & 0xff) -
                 (Platform.getByte(rightObj, rightOff + i) & 0xff);
-        if (res != 0) return res;
+        if (res != 0) return (int) res;
```

Comment:
How about the following change to minimize and localize the change?
```
int res = (Platform.getByte(leftObj, leftOff + i) & 0xff) -
             (Platform.getByte(rightObj, rightOff + i) & 0xff);
if (res != 0) return res;
```

Reply:
We can restrict scope of 'res' as an 'int' : and avoid the type promotions/conversions.